### PR TITLE
fix(*): Fix various bugs

### LINKED ISSUE #552: "Copy Cloud Link" not working on mac
Clipboard does not show the cloud link. 
c.f https://github.com/qri-io/desktop/blob/21c81f2b0c0f0a21df7b0ec4f1d8a5629a362b45/app/components/workbench/headerButtons/PublishButton.tsx#L47

### LINKED ISSUE #591: History list overflows container when viewing a dataset in the network tab
### What is your OS and version?
MacOS Catalina 10.15.5 (19F101)

### What version of qri are you using (`qri version`)?
0.9.12-dev

### What version of Qri Desktop are you using?
0.4.5

## Issue

History list overflows container when viewing a dataset in the network tab.  The panel that contains the list should be scrollable.  Instead the containing element is scrollable, causing a weird display state.

Qri_Desktop

### LINKED ISSUE #592: bug(rename): desktop doesn't allow `-`, but qri core does
Should adjust the validator to allow `-`

https://github.com/qri-io/desktop/blob/981fadc4ba41e38f0fedf4a4470dd8dc0765afa1/app/utils/formValidation.ts#L49

Need to add `-` to this regex:

```
 const invalidCharacters = !(/^(?![0-9])[a-z0-9_-]+$/.test(name))
```

### LINKED ISSUE #595: No padding below commit button
Qri_Desktop

### LINKED ISSUE #596: Pasting text copied from website into description fields results in html
Copying rich text from a website and pasting it into a schema description text input causes the input to contain html.

The workaround has been pasting into a text editor, then copying and pasting again into Qri Desktop.

### LINKED ISSUE #597: Nonfunctional tooltip icons on Commit pane
Qri_Desktop

### PR DESCRIPTION
**Pull request by ramfox**
closes #597 
closes #596 
closes #595 
closes #552 
closes #591 
closes #592

#593, not able to reproduce the error found by a beta user on windows. Asked for some help from @dustmop or @b5 to see if it is a windows only bug

Going to push #593 to a future pr, if we can repro it

### CODE REVIEW

**Comment by ramfox** [comment]
@chriswhong it would be great if you could build desktop off this branch and try to repro any of the bugs on this list you filed. The major one would be the "paste into schema description" bug

File: app/components/workbench/components/CommitEditor.tsx [hunk 1]
```diff
@@ -43,6 +44,10 @@ export const CommitEditorComponent: React.FunctionComponent<CommitEditorProps> =
 
   const { username, name } = qriRef
 
+  React.useEffect(() => {
+    ReactTooltip.rebuild()
+  }, [])
```
Comment (b5) [inline app/components/workbench/components/CommitEditor.tsx:49]: Unless this function _must_ be bound to the current context, this should work and be a little cleaner:
```suggestion
  React.useEffect(ReactTooltip.rebuild, [])
```

While we're here can we get a basic comment on what this does? I'm guessing tooltips need to be rebuilt on render, but is there a specific reason? data-model specific stuff in tooltips?

File: app/components/workbench/headerButtons/PublishButton.tsx [hunk 2]
```diff
@@ -32,6 +33,10 @@ const PublishButtonComponent: React.FunctionComponent<PublishButtonProps> = (pro
     setModal
   } = props
 
+  React.useEffect(() => {
+    ReactTooltip.rebuild()
+  }, [])
```
Comment (b5) [inline app/components/workbench/headerButtons/PublishButton.tsx:38]: ditto on the refactoring comment

**Review by b5 — APPROVED** [review]
one nit repeated in two places, otherwise LGTM!

**Review by chriswhong — APPROVED** [review]
Ran locally and reviewed each bug, everything is looking great!

File: app/components/workbench/components/CommitEditor.tsx (hunk 1 above)
Comment (ramfox) [reply app/components/workbench/components/CommitEditor.tsx:49]: We build the tooltips in the App component. Anything that is not rendered in the first render cycle won't be binded to the ReactTooltip component. So on first render of the component, we rebuild the tooltips.

I'll add the comment!

**Pull request merged**